### PR TITLE
Feature/data encapsulation

### PR DESCRIPTION
**Pull request by minsulee2**
As a format of messages in p2p communication has almost been fixed, the messages seem to be encapsulated.

Changes:
- data encapsulation with fixed format.
- Tidy up some codes

### CODE REVIEW

**Comment by platfowner** [comment]
@minsulee2 Could you add some PR description? (what it does and why this is needed?)

File: p2p/util.js [hunk 1]
```diff
@@ -132,6 +135,24 @@ function checkProtoVer(connections, socket, minProtocolVersion, maxProtocolVersi
   return true;
 }
 
+function encapsulateMessage(type, dataObj) {
+  if (!type) {
+    logger.error('Type must be specified.');
```
Comment (platfowner) [inline p2p/util.js:140]: minor: a separate variable for the message?

File: p2p/util.js [hunk 2]
```diff
@@ -132,6 +135,24 @@ function checkProtoVer(connections, socket, minProtocolVersion, maxProtocolVersi
...
+    throw new Error('Type must be specified.');
+  };
+  if (!dataObj) {
+    logger.error('dataObj cannot be null or undefined.');
```
Comment (platfowner) [inline p2p/util.js:144]: ditto

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Please make sure all tests pass, thanks!

File: p2p/util.js [hunk 3]
```diff
@@ -132,6 +135,24 @@ function checkProtoVer(connections, socket, minProtocolVersion, maxProtocolVersi
   return true;
 }
 
+function encapsulateMessage(type, dataObj) {
+  if (!type) {
+    logger.error('Type must be specified.');
+    throw new Error('Type must be specified.');
```
Comment (liayoo) [inline p2p/util.js:141]: Could we return null instead of throwing errors here and below?

File: p2p/util.js (hunk 1 above)
Comment (minsulee2) [reply p2p/util.js:140]: Done! As it returns null rather than throwing error, no need to make the variable. Thanks :)

File: p2p/util.js (hunk 2 above)
Comment (minsulee2) [reply p2p/util.js:144]: Done!

File: p2p/util.js (hunk 3 above)
Comment (minsulee2) [reply p2p/util.js]: Done!

File: common/constants.js [hunk 4]
```diff
@@ -52,7 +52,7 @@ if (!fs.existsSync(PROTOCOL_VERSIONS)) {
   throw Error('Missing protocol versions file: ' + PROTOCOL_VERSIONS);
 }
 const PROTOCOL_VERSION_MAP = JSON.parse(fs.readFileSync(PROTOCOL_VERSIONS));
-const DATA_PROTOCOL_VERSION = "1.0.0";
+const DATA_PROTOCOL_VERSION = "1.1.0";
```
Comment (minsulee2) [inline common/constants.js:55]: @platfowner @liayoo As some changes are made, updated to 1.1.0. The next item can be version compatibility for semver.lt() parts to compatible with 1.0.0 and 1.1.0.

**Comment by minsulee2** [comment]
@liayoo PTAL

File: common/constants.js (hunk 4 above)
Comment (liayoo) [reply common/constants.js:55]: 👍

**Review by liayoo — APPROVED** [review]
LGTM, thanks!

**Comment by minsulee2** [comment]
I will move on to next stuff. Thanks!!

**Pull request merged**